internal/events/events.go

```diff
@@ -124,17 +124,28 @@ func eventString(writer io.Writer, event flow.Event) {
 		printField(writer, field, value)
 	}
 }
-
+func printValues(writer io.Writer, fieldIdentifier, typedId, valueString string) {
+	_, _ = fmt.Fprintf(writer, "\t\t- %s (%s): %s \n", fieldIdentifier, typedId, valueString)
+}
 func printField(writer io.Writer, field cadence.Field, value cadence.Value) {
+	v := value.String()
 	var typeId string
+
+	defer func() {
+		if err := recover(); err != nil {
+			printValues(writer, field.Identifier, "?", v)
+		}
+	}()
+
 	if field.Type != nil {
+		//TODO: onflow/cadence issue #1672
+		//currently getting ID for cadence array will cause panic
 		typeId = field.Type.ID()
 	}
 
-	v := value.String()
 	if typeId == "" { // exception for not known typeId workaround for cadence arrays
 		v = fmt.Sprintf("%s\n\t\thex: %x", v, v)
+		typeId = "?"
 	}
-
-	_, _ = fmt.Fprintf(writer, "\t\t- %s (%s): %s \n", field.Identifier, typeId, v)
+	printValues(writer, field.Identifier, typeId, v)
```

Comment:
Tested, for argument of optional type like [String]? output is 

````
Events:          
    Index       0
    Type        A.f8d6e0586b0a20c7.HelloWorld.FooEvent
    Tx ID       eee4f24c6958a8978bacf2c0585608d5678b1d867e8dec3961e0a726ea714b5c
    Values
                - x (?): ["Event"] 

````

for non optional type like [String] output is 

````
Events:          
    Index       0
    Type        A.f8d6e0586b0a20c7.HelloWorld.FooEvent
    Tx ID       794e1ac2a5c5add70075e4b162bb84f15f61d7b5ac1ff560187d4d96d5e4b4c3
    Values
                - x (?): ["Event"]
                hex: 5b224576656e74225d 
````